client/client_test.go

```diff
@@ -391,15 +391,15 @@ func assertRepoHasExpectedKeys(t *testing.T, repo *NotaryRepository,
 	assert.NoError(t, err)
 
 	roles := make(map[string]bool)
-	for keyID, role := range ks.ListKeys() {
-		if role == data.CanonicalRootRole {
+	for keyID, keyInfo := range ks.ListKeys() {
+		if keyInfo.Role == data.CanonicalRootRole {
 			assert.Equal(t, rootKeyID, keyID, "Unexpected root key ID")
 		}
 		// just to ensure the content of the key files created are valid
 		_, r, err := ks.GetKey(keyID)
 		assert.NoError(t, err)
-		assert.Equal(t, role, r)
-		roles[role] = true
+		assert.Equal(t, keyInfo.Role, r)
+		roles[keyInfo.Role] = true
```

Comment:
No, since each `keyInfo` in `ListKeys` must be tied to a keyID, and when we remove any `KeyInfo` the whole entry is deleted from `keyInfoMap`